### PR TITLE
Site Editor: Move the save view state to the edit site store

### PR DESCRIPTION
**Pull request by youknowriad**
Extracted from #44770

## What?

Instead of using local state for the save view state, we move this to the edit site store.

## Why?

This allows us to allow unnecessary props passing top down, which is also great to make our components more reusable.

## How?

Just did the same thing we do for list view state.

## Testing Instructions

Make some changes in the site editor and save them.

### CODE REVIEW

File: packages/edit-site/src/store/reducer.js [hunk 1]
```diff
@@ -183,6 +183,26 @@ export function listViewPanel( state = false, action ) {
...
 }
 
+/**
+ * Reducer to set the save view panel open or closed.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ */
+export function saveViewPanel( state = false, action ) {
+	switch ( action.type ) {
```
Comment (ntsekouras) [inline packages/edit-site/src/store/reducer.js:193]: haven't tested yet, but should we include `SET_IS_SAVE_VIEW_OPENED` to other reducers like `listViewPanel`, etc..?

File: packages/edit-site/src/store/reducer.js [hunk 2]
```diff
@@ -183,6 +183,26 @@ export function listViewPanel( state = false, action ) {
...
+		case 'OPEN_NAVIGATION_PANEL_TO_MENU':
+			return false;
+		case 'SET_IS_NAVIGATION_PANEL_OPENED':
+			return action.isOpen ? false : state;
+		case 'SET_IS_SAVE_VIEW_OPENED':
+			return action.isOpen;
+		case 'SET_CANVAS_MODE':
```
Comment (youknowriad) [inline packages/edit-site/src/store/reducer.js]: This should be removed :)

File: packages/edit-site/src/store/reducer.js (hunk 1 above)
Comment (youknowriad) [reply packages/edit-site/src/store/reducer.js:193]: No, I don't think so, for the listViewPanel reducer, there's a rule that says "inserter" and "list view" can't be opened at the same time, it's not the case here.

File: packages/edit-site/src/components/editor/index.js [hunk 3]
```diff
@@ -286,28 +277,28 @@ function Editor( { onError } ) {
...
 														/>
 													) : (
 														<div className="edit-site-editor__toggle-save-panel">
 															<Button
 																variant="secondary"
 																className="edit-site-editor__toggle-save-panel-button"
-																onClick={
-																	openEntitiesSavedStates
+																onClick={ () =>
+																	setIsInserterOpened(
```
Comment (ntsekouras) [inline packages/edit-site/src/components/editor/index.js]: Shouldn't this be `setIsSaveViewOpened`?

Comment (youknowriad) [reply packages/edit-site/src/components/editor/index.js]: opus :)

**Review by ntsekouras — APPROVED** [review]
LGTM, thanks!

**Comment by Mamaduka** [comment]
This is a nice cleanup!

**Pull request merged**